### ISSUE TITLE
Android application dies when I send notifications from Pinpoint and never receive a push notification.

### ISSUE THREAD

**Issue by jabzumnd** [aws-amplify/amplify-js#3538]
**Describe the bug**
I have an application already made in React Native and I want to integrate sending push notifications with Amplify. When I send a test message from Pinpoint with the generated token, the application stops and never receives the push notification.

Creating an application from scratch works everything correctly, but with my existing application it does not work. I followed the steps of the Amplify documentation in both scenarios and only work in this one (creating from 0 the application).

**To Reproduce**
1. Follow the steps of Amplify Push Notifications: https://aws-amplify.github.io/docs/js/push-notifications
2. Send a test message from Pinpoint in the AWS console.

**Expected behavior**
Receive the push notification of my application.

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy S5

**Sample code**
My App.js file in React Native is:
```
import React, { Component } from "react";
import {
  Platform,
  AsyncStorage,
  StyleSheet,
  Text,
  View,
  PushNotificationIOS
} from "react-native";
import SplashScreen from "react-native-splash-screen";
import Analytics from "@aws-amplify/analytics";
import PushNotification from "@aws-amplify/pushnotification";
import aws_exports from "./aws-exports";

// PushNotification need to work with Analytics
Analytics.configure(aws_exports);

PushNotification.configure(aws_exports);

type Props = {};
export default class App extends Component<Props> {
  state = {
    token: "xd"
  };
  componentDidMount() {
    SplashScreen.hide();
    try {
      // get the notification data when notification is received
      PushNotification.onNotification(notification => {
        // Note that the notification object structure is different from Android and IOS
        console.log("in app notification", notification);

        try {
          // required on iOS only (see fetchCompletionHandler docs: https://facebook.github.io/react-native/docs/pushnotificationios.html)
          notification.finish(PushNotificationIOS.FetchResult.NoData);
        } catch (e) {
          console.log("error al finalizar en ios");
        }
      });

      // get the registration token
      PushNotification.onRegister(token => {
        console.log("in app registration", token);
        this.setState({ token: token });
      });

      // get the notification data when notification is opened
      PushNotification.onNotificationOpened(notification => {
        console.log("the notification is opened", notification);
      });
    } catch (e) {
      console.log("error xd", e);
    }

    AsyncStorage.getAllKeys().then(keys =>
      AsyncStorage.multiGet(keys).then(result => {
        result.map(req =>
          req.forEach(element => {
            this.setState({ favorites: element });
            console.log(this.state.favorites);
          })
        );
      })
    );
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to React Native!</Text>
        <Text selectable style={styles.welcome}>
         Token: {this.state.token}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF"
  },
  welcome: {
    fontSize: 20,
    textAlign: "center",
    margin: 10
  },
  instructions: {
    textAlign: "center",
    color: "#333333",
    marginBottom: 5
  }
});

```

These are my configuration files:

android/build.gradle

```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    ext {
        buildToolsVersion = "28.0.3"
        minSdkVersion = 16
        compileSdkVersion = 29
        targetSdkVersion = 28
        supportLibVersion = "28.0.0"
        googlePlayServicesAuthVersion = "15.0.1" // <--- use this version or newer
    }
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.2.1'
        //classpath 'com.google.gms:google-services:4.0.1' // <--- use this version or newer
        // classpath 'com.google.gms:google-services:4.2.0'
        classpath 'com.google.gms:google-services:+'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
    subprojects {
        afterEvaluate { project ->
            if (project.hasProperty("android")) {
                android {
                    compileSdkVersion rootProject.ext.compileSdkVersion
                    buildToolsVersion rootProject.ext.buildToolsVersion
                }
            }
        }
    }
}

allprojects {
    repositories {
        mavenLocal()
        google()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
        maven {
            url "https://maven.google.com"
        }
    }
}


task wrapper(type: Wrapper) {
    gradleVersion = '4.7'
    distributionUrl = distributionUrl.replace("bin", "all")
}

```

android/app/build.gradle

```
.
.
.
.
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    buildToolsVersion '29.0.0'

    defaultConfig {
        applicationId "com.app_in_production"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 5
        versionName "0.18"
    }
    signingConfigs {
        debug {
            storeFile file('keystore/debug.keystore')
            keyAlias '####'
            keyPassword '###'
            storePassword '###'
        }
        release{
            if (project.hasProperty('MYAPP_RELEASE_STORE_FILE')) {
                storeFile file(MYAPP_RELEASE_STORE_FILE)
                storePassword MYAPP_RELEASE_STORE_PASSWORD
                keyAlias MYAPP_RELEASE_KEY_ALIAS
                keyPassword MYAPP_RELEASE_KEY_PASSWORD
            }
        }
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86", "arm64-v8a"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2, "arm64-v8a": 3]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    implementation project(':amazon-cognito-identity-js')
    implementation project(':@aws-amplify_pushnotification')
    implementation project(':react-native-webview')
    implementation project(':react-native-google-signin')
    implementation project(':react-native-splash-screen')
    implementation project(':react-native-keychain')
    implementation project(':react-native-gesture-handler')
    implementation project(':react-native-vector-icons')
    implementation fileTree(include: ['*.jar'], dir: 'libs')
    implementation "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
    implementation 'com.facebook.react:react-native:+'
    // From node_modules
    implementation 'com.google.firebase:firebase-core:16.0.1'
    implementation 'com.google.firebase:firebase-messaging:12.0.1'
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

apply plugin: 'com.google.gms.google-services' // <--- this should be the last line
```

package.json
```
{
  "name": "APP_IN_PRODUCTION",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "@aws-amplify/pushnotification": "^1.0.28",
    "accordion-collapse-react-native": "^0.1.9",
    "aws-amplify": "^1.1.29",
    "axios": "^0.18.0",
    "native-base": "^2.12.1",
    "prop-types": "^15.6.2",
    "react": "16.6.3",
    "react-content-loader": "^4.2.1",
    "react-native": "0.58.2",
    "react-native-elements": "^0.19.1",
    "react-native-gesture-handler": "^1.0.15",
    "react-native-google-signin": "^1.1.0",
    "react-native-keychain": "^3.0.0",
    "react-native-pure-chart": "0.0.24",
    "react-native-render-html": "^4.1.2",
    "react-native-splash-screen": "^3.0.6",
    "react-native-table-component": "^1.2.0",
    "react-native-vector-icons": "^6.1.0",
    "react-native-webview": "^5.2.3",
    "react-navigation": "^3.0.9"
  },
  "devDependencies": {
    "babel-core": "7.0.0-bridge.0",
    "babel-jest": "24.0.0",
    "jest": "24.0.0",
    "metro-react-native-babel-preset": "0.51.1",
    "react-test-renderer": "16.6.3"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

AndroidManifest.xml

```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    package="com.app_in_production">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>

    <application
        android:name=".MainApplication"
        android:label="@string/app_name"
        android:icon="@mipmap/ic_launcher"
        android:allowBackup="false"
        android:theme="@style/AppTheme"
        tools:replace="android:allowBackup">

        <activity
            android:name=".SplashActivity"
            android:theme="@style/SplashTheme"
            android:label="@string/app_name">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>

        <activity
            android:name=".MainActivity"
            android:label="@string/app_name"
            android:configChanges="keyboard|keyboardHidden|orientation|screenSize"      
            android:windowSoftInputMode="stateAlwaysHidden|adjustPan"
            android:exported="true">
        </activity>
        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

        <!--[START Push notification config -->
        <!-- [START firebase_service] -->
        <service
            android:name="com.amazonaws.amplify.pushnotification.RNPushNotificationMessagingService">
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT"/>
            </intent-filter>
        </service>
        <!-- [END firebase_service] -->
        <!-- [START firebase_iid_service] -->
        <service
            android:name="com.amazonaws.amplify.pushnotification.RNPushNotificationDeviceIDService">
            <intent-filter>
                <action android:name="com.google.firebase.INSTANCE_ID_EVENT"/>
            </intent-filter>
        </service>
        <receiver
            android:name="com.amazonaws.amplify.pushnotification.modules.RNPushNotificationBroadcastReceiver"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.amazonaws.amplify.pushnotification.NOTIFICATION_OPENED"/>
            </intent-filter>
        </receiver>
        <!-- [END Push notification config -->

    </application>

</manifest>

```

**Comment by powerful23** [comment]
@jabzumnd was there any error logged in the `logcat` when your app crashed?

**Comment by jabzumnd** [comment]
Hi @powerful23  I solved the issue, I did tests on a newer phone that gave me the error:

java.lang.abstractmethoderror: abstract method void com.google.firebase.iid-zzb.zzd (android.content.intent) unknow source: 26

My Samsung Galaxy S5 or the android version not helpful at all, because I could not visualize this error all yesterday.

So, the issue was because my old configuration in the android/app/build.gradle was

`implementation 'com.google.firebase:firebase-core:16.0.1'
implementation 'com.google.firebase:firebase-messaging:12.0.1'`

The problem solves when I update the firebase-messaging version to:

`implementation 'com.google.firebase:firebase-core:16.0.1'
implementation 'com.google.firebase:firebase-messaging:17.4.0'`

You can view all the versions in: https://maven.google.com

In android the token can be visualized only once when PushNotification.onRegister is executed so I will keep this token in the storage with AsyncStorage so that I can manipulate it or when change it to update it in my BD.

**Comment by psyrenpark** [comment]
thanks

**Comment by jordanranz** [comment]
The original posted issue seems to have been solved here. Please open a new issue if you are experiencing any related problems.
